File: examples/python/best-practices/avoid-sequential-operations.py
@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> dict:
    # Sequential execution - each step waits for the previous one
    user_data = context.step(fetch_user_data(event["user_id"]))
    order_history = context.step(fetch_order_history(event["user_id"]))
    preferences = context.step(fetch_preferences(event["user_id"]))
    
    return {
        "user": user_data,
        "orders": order_history,
        "preferences": preferences,
    }

File: examples/java/best-practices/avoid-sequential-operations.java
var userId = (String) input.get("user_id");
// Sequential execution - each step waits for the previous one
var userData = ctx.step("fetch_user_data", Map.class, stepCtx -> fetchUserData(userId));
var orderHistory = ctx.step("fetch_order_history", Map.class, stepCtx -> fetchOrderHistory(userId));
var preferences = ctx.step("fetch_preferences", Map.class, stepCtx -> fetchPreferences(userId));

return Map.of(
    "user", userData,
    "orders", orderHistory,
    "preferences", preferences
);
